zrml/prediction-markets/src/lib.rs

```diff
@@ -2606,40 +2567,45 @@ mod pallet {
                 }
             }
 
-            for (i, dispute) in disputes.iter().enumerate() {
-                let actual_bond = default_dispute_bond::<T>(i);
-                if dispute.outcome == resolved_outcome {
-                    T::AssetManager::unreserve_named(
-                        &Self::reserve_id(),
-                        Asset::Ztg,
-                        &dispute.by,
-                        actual_bond,
-                    );
-
-                    correct_reporters.push(dispute.by.clone());
-                } else {
-                    let (imbalance, _) = CurrencyOf::<T>::slash_reserved_named(
-                        &Self::reserve_id(),
-                        &dispute.by,
-                        actual_bond.saturated_into::<u128>().saturated_into(),
-                    );
-                    overall_imbalance.subsume(imbalance);
+            let mut correct_disputor = None;
+            if let Some(bond) = market.bonds.dispute.clone() {
+                if !bond.is_settled {
+                    if is_correct {
+                        let imb = Self::slash_dispute_bond(market_id, None)?;
+                        overall_imbalance.subsume(imb);
+                    } else {
+                        // If the report outcome was wrong, the dispute was justified
+                        Self::unreserve_dispute_bond(market_id)?;
+                        correct_disputor = Some(bond.who);
+                    }
                 }
             }
 
-            // Fold all the imbalances into one and reward the correct reporters. The
-            // number of correct reporters might be zero if the market defaults to the
-            // report after abandoned dispute. In that case, the rewards remain slashed.
-            if let Some(reward_per_each) =
-                overall_imbalance.peek().checked_div(&correct_reporters.len().saturated_into())
-            {
-                for correct_reporter in &correct_reporters {
-                    let (actual_reward, leftover) = overall_imbalance.split(reward_per_each);
-                    overall_imbalance = leftover;
-                    CurrencyOf::<T>::resolve_creating(correct_reporter, actual_reward);
-                }
+            let mut imbalance_left = <NegativeImbalanceOf<T>>::zero();
+            if let Some(disputor) = correct_disputor {
+                // TODO: The disputor gets at most the outsider and oracle bond. Is okay?
+                CurrencyOf::<T>::resolve_creating(&disputor, overall_imbalance);
+            } else {
+                // TODO: The MDMs get at most the slashed dispute bond. Is okay?
```

Comment:
I think both of those TODOs are fine.

Reply:
c3c5ca6baee310aa05a7c99758db614d372f988c